plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/callables/FullNamedFunctionSignature.kt

```diff
@@ -19,6 +24,28 @@ interface FullNamedFunctionSignature : NamedFunctionSignature {
     val declarationSource: KtSourceElement?
 }
 
+interface PrimaryConstructorFunctionSignature : FullNamedFunctionSignature {
+
+    private fun primaryConstructorFieldsWithParams(): List<Pair<PrimaryConstructorFieldEmbedding, VariableEmbedding>> {
+        val fields = (returnType as? ClassTypeEmbedding)?.fields?.toList() ?: return emptyList()
+        //second filter is basically cast and doesn't check anything
+        return fields.map { (name, field) ->
+            val correspondingParam = params.find { ScopedKotlinName(ParameterScope, name) == it.name }
```

Comment:
I'm wondering whether there's a neater way to do this.  After all, at this point we should know which parameters are actually fields—could we generate a list from that directly, rather than cross-searching fields and parameters?

Reply:
Another thing we could do though is to create a separate instance of `VariableEmbedding` with the same name.

Reply:
I think that we should already be getting `FirVariableEmbeddings` here. I'm not sure whether they have the right symbol, but if not I think we should get access to that somehow.